src/main/java/org/alfresco/utility/data/DataContent.java

```diff
@@ -189,57 +303,139 @@ public String addEmailAlias(String alias) throws Exception
 
-        if (getLastResource().isEmpty())
-            setLastResource(RandomData.getRandomName("Folder"));
+        // Set Title or Description if specified
+        if (fileModel.getTitle() != null || fileModel.getDescription() != null)
+        {
+            body.put("cm:title", fileModel.getTitle());
+            body.put("cm:description", fileModel.getDescription());
+        }
+
+        post.setEntity(client.setMessageBody(body));
+
+        // Send Request
+        logger.info(String.format("Create content with name '%s' by: ", fileModel.getName()));
+        logger.info(String.format("POST: '%s'", reqUrl));
+        HttpResponse response = client.execute(currentUser.getUsername(), currentUser.getPassword(), post);
+        if(HttpStatus.SC_CREATED == response.getStatusLine().getStatusCode())
+        {
+            JSONObject entryResponse = client.readStream(response.getEntity());
+            JSONObject entryValueMap = (JSONObject) entryResponse.get("entry");
+
+            fileModel.setNodeRef(entryValueMap.get("id").toString());
+            fileModel.setName(entryValueMap.get("name").toString());
+
+            String fileLocation = Utility.buildPath(getLastResource(), fileModel.getName());
+            fileModel.setCmisLocation(fileLocation);
+            fileModel.setProtocolLocation(fileLocation);
+
+            logger.info(String.format("Successful created content with id '%s' ", entryValueMap.get("id").toString()));
+            return fileModel;
+        }
+        else
+        {
+            logger.error(client.getParameterFromJSON(response,"briefSummary", "error"));
+            return new FileModel();
+        }
+    }
+
+    /**
+     * This method updates the content of the created document
+     *
+     * @param client
+     * @param fileModel
+     */
+    public void updateContent(AlfrescoHttpClient client, FileModel fileModel)
+    {
+        // Build request
+        String nodeId = fileModel.getNodeRef();
+        String reqUrl = client.getApiVersionUrl() + "nodes/" + nodeId + "/content?majorVersion=true";
+
+        HttpPut put  = new HttpPut(reqUrl);
+        String fileContent = fileModel.getContent();
+        StringEntity se = new StringEntity(fileContent, client.UTF_8_ENCODING);
+        se.setContentType(new BasicHeader(HTTP.CONTENT_TYPE, client.MIME_TYPE_JSON));
+        put.setEntity(se);
 
-        Document cmisDocument = null;
         try
         {
-            cmisDocument = contentService.createDocumentInRepository(getSession(), getLastResource(), documentType, newContent, "This is a file file");
+            // Send Request
+            logger.info(String.format("Update content for file with name '%s' by: ", fileModel.getName()));
+            logger.info(String.format("POST: '%s'", reqUrl));
+            HttpResponse response = client.executeAndRelease(currentUser.getUsername(), currentUser.getPassword(), put);
+            if(HttpStatus.SC_OK == response.getStatusLine().getStatusCode())

```

Comment:
what happens if this is not OK ? 
do the calling methods have enough information to handle this case correctly ?